Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs

```diff
@@ -52,46 +52,91 @@ public enum Configurations
             IOSCameraUsageDescription,
         };
 
+        private class ConfigGetter
+        {
+            /// <summary>
+            /// Array of build targets where the get action is valid
+            /// </summary>
+            public BuildTarget[] ValidTargets;
+
+            /// <summary>
+            /// Action to perform to determine if the current configuration is correctly enabled
+            /// </summary>
+            public Func<bool> GetAction;
+
+            public ConfigGetter(Func<bool> get, BuildTarget target = BuildTarget.NoTarget)
+            {
+                GetAction = get;
+                ValidTargets = new BuildTarget[] { target };
+            }
+
+            public ConfigGetter(Func<bool> get, BuildTarget[] targets)
+            {
+                GetAction = get;
+                ValidTargets = targets;
+            }
+
+            /// <summary>
+            /// Returns true if the active build target is contained in the ValidTargets list or the list contains a BuildTarget.NoTarget entry, false otherwise
+            /// </summary>
+            public bool IsActiveBuildTargetValid()
+            {
+                foreach (var buildTarget in ValidTargets)
+                {
+                    if (buildTarget == BuildTarget.NoTarget || buildTarget == EditorUserBuildSettings.activeBuildTarget)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
         // The check functions for each type of setting
-        private static readonly Dictionary<Configurations, Func<bool>> ConfigurationGetters = new Dictionary<Configurations, Func<bool>>()
+        private static readonly Dictionary<Configurations, ConfigGetter> ConfigurationGetters = new Dictionary<Configurations, ConfigGetter>()
         {
-            { Configurations.LatestScriptingRuntime,  () => { return IsLatestScriptingRuntime(); } },
-            { Configurations.ForceTextSerialization,  () => { return IsForceTextSerialization(); } },
-            { Configurations.VisibleMetaFiles,  () => { return IsVisibleMetaFiles(); } },
+            { Configurations.LatestScriptingRuntime, new ConfigGetter(() => { return IsLatestScriptingRuntime(); }) },
+            { Configurations.ForceTextSerialization, new ConfigGetter(() => { return IsForceTextSerialization(); }) },
+            { Configurations.VisibleMetaFiles, new ConfigGetter(() => { return IsVisibleMetaFiles(); }) },
             // Ensure compatibility with the pre-2019.3 XR architecture for customers / platforms
             // with legacy requirements.
 #pragma warning disable 0618
-            { Configurations.VirtualRealitySupported,  () => { return PlayerSettings.virtualRealitySupported; } },
+            { Configurations.VirtualRealitySupported, new ConfigGetter(() => { return PlayerSettings.virtualRealitySupported; }) },
```

Comment:
can you update this to use the new XRSettingsUtilities.LegacyXRSupported property?

Reply:
that will allow removing the pragma

Reply:
ah yep yep

Reply:
Updated